### PR TITLE
Added skiko events for android.

### PR DESCRIPTION
**Pull request by Rsedaikin**
- skiko events (raw touches, gestures, keyboard, typing) for android platform
- added [doubletap] gesture recognition
- fixes for android rendering pipeline
- refactored [Clocks] example due to emulator shortcuts

### CODE REVIEW

File: samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/Clocks.kt [hunk 1]
```diff
@@ -152,20 +159,23 @@ class Clocks(private val layer: SkiaLayer): SkikoView {
             when (event.key) {
                 SkikoKey.KEY_NUMPAD_ADD -> scale *= 1.1
                 SkikoKey.KEY_I -> {
-                    if (event.modifiers == SkikoInputModifiers.CONTROL) {
+                    if (event.modifiers == SkikoInputModifiers.SHIFT) {
```
Comment (olonho) [inline samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/Clocks.kt:162]: Why this change?

File: samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/MainActivity.kt [hunk 2]
```diff
@@ -19,7 +21,7 @@ class MainActivity : Activity() {
             //layoutParams = ViewGroup.LayoutParams(1000, 1200)
         }
 
-        val skiaLayer = SkiaLayer()
+        val skiaLayer = SkiaLayer(SkikoGestureEventKind.values())
```
Comment (olonho) [inline samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/MainActivity.kt:24]: Hmm, not sure if like this change. Skia layer shalln't be inited this way.

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt [hunk 3]
```diff
@@ -29,6 +47,276 @@ class SkikoSurfaceView(context: Context, layer: SkiaLayer) : GLSurfaceView(conte
     fun scheduleFrame() {
         frameDispatcher.scheduleFrame()
     }
+
+    private val simpleGestures = GestureDetector(context, object: SimpleOnGestureListener() {
+        override fun onSingleTapUp(event: MotionEvent): Boolean {
+            if (!layer.gestures!!.contains(SkikoGestureEventKind.TAP)) return false
```
Comment (olonho) [inline skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt:53]: `!!` looks like not the best idea.

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt [hunk 4]
```diff
@@ -29,6 +47,276 @@ class SkikoSurfaceView(context: Context, layer: SkiaLayer) : GLSurfaceView(conte
...
+                    y = (event.y / density).toDouble(),
+                    kind = SkikoGestureEventKind.TAP,
+                    platform = event
+                )
+            )
+            return true
+        }
+
+        override fun onDoubleTap(event: MotionEvent): Boolean {
+            if (!layer.gestures!!.contains(SkikoGestureEventKind.DOUBLETAP)) return false
```
Comment (olonho) [inline skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt:67]: Same

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt [hunk 5]
```diff
@@ -29,6 +47,276 @@ class SkikoSurfaceView(context: Context, layer: SkiaLayer) : GLSurfaceView(conte
     fun scheduleFrame() {
         frameDispatcher.scheduleFrame()
     }
+
+    private val simpleGestures = GestureDetector(context, object: SimpleOnGestureListener() {
```
Comment (olonho) [inline skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt:51]: Not sure if this whole logic belongs to Renderer.android.kt, better move to separate file.

**Comment by olonho** [comment]
Overall looks and behaves pretty good, need some code and API cleanups.

File: samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/Clocks.kt (hunk 1 above)
Comment (Rsedaikin) [reply samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/Clocks.kt:162]: > refactored [Clocks] example due to emulator shortcuts
Android emulator uses CONTROL key to emulate two-finger-touch

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt (hunk 5 above)
Comment (Rsedaikin) [reply skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt]: Done.

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt (hunk 4 above)
Comment (Rsedaikin) [reply skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt]: Done.

File: skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt (hunk 3 above)
Comment (Rsedaikin) [reply skiko/src/androidMain/kotlin/org/jetbrains/skiko/Renderer.android.kt]: Done.

File: samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/MainActivity.kt (hunk 2 above)
Comment (olonho) [reply samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/MainActivity.kt:24]: Maybe smth like `layer.eventsToListen = arrayOf(...)`?

Comment (Rsedaikin) [reply samples/SkiaAndroidSample/src/main/java/org/jetbrains/skiko/sample/MainActivity.kt:24]: Maybe it's a good idea indeed, I'm developing it now and also will try this approach.

**Review by olonho — APPROVED** [review]
